### PR TITLE
np.packbits and np.unpackbits

### PR DESCRIPTION
**Pull request by magnatelee**

### CODE REVIEW

File: cunumeric/bits.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    numpy.packbits
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    a, sanitized_axis = _sanitize_arguments(a, axis, bitorder)
+
+    if a.dtype.kind not in ("u", "i", "b"):
```
Comment (ipdemes) [inline cunumeric/bits.py:92]: I think you can move this check before `_sinitize_arguments` is called so the TypeError is caught before anything else

File: cunumeric/bits.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    if a.dtype != "B":
+        raise TypeError("Expected an input array of unsigned byte data type")
+
+    if count is None:
+        sanitized_count = a.shape[sanitized_axis] * 8
+    else:
+        if not isinstance(count, int):
+            raise TypeError("count must be an integer")
+        sanitized_count = count
```
Comment (ipdemes) [inline cunumeric/bits.py:169]: these else logic can be replaced with the line 172 since count is not supported in this implementation

**Comment by ipdemes** [comment]
I think you might want to add to the test 2 following use cases:

1) input array is empty

2) result of unpackits is used for packbits like below:

```

 shape = (5,2,10)
    arr = np.random.randint(low=0, high=255, size=shape, dtype="B")
    out_np = np.unpackbits(arr, axis =2)
    arr_num = num.array(arr)
    out_num = num.unpackbits(arr_num, axis =2)
    assert np.array_equal(out_np, out_num)

    out_np_2 = np.packbits(out_np, axis =2)
    out_num_2 = np.packbits(out_num, axis =2)
    assert np.array_equal(out_np_2, out_num_2)

    assert np.array_equal(out_np_2, arr)
    assert np.array_equal(out_num_2, arr_num)
    
    ```

File: cunumeric/bits.py (hunk 1 above)
Comment (magnatelee) [reply cunumeric/bits.py:90]: 50223d236aa3292803049592c5390bebbcac5579

File: cunumeric/bits.py (hunk 2 above)
Comment (magnatelee) [reply cunumeric/bits.py]: `count` is now handled correctly (4cd1a68304b165619d211a3389b04ea197c270e7)

**Comment by magnatelee** [comment]
> I think you might want to add to the test 2 following use cases:

I added tests for the first case, but not for the second one, as I don't think it's necessary; the NumPy implementation already satisfies the roundtrip property, so checking the equivalence between that and the cuNumeric implementation should be sufficient.

**Review by ipdemes — APPROVED** [review]

**Pull request merged**